File: languages/php/Bloock/Entity/Availability/PublishResponse.php
<?php

namespace Bloock\Entity\Availability;

use Exception;

/**
 * Represents an object with a the publish response attributes.
 */
class PublishResponse
{
    public string $id;
    public ?IpnsKey $ipnsKey = null;

    /**
     * Constructs a PublishResponse object with the specified parameters.
     * @param $id
     * @param $ipnsKey
     * @throws Exception
     */
    public function __construct(string $id, ?IpnsKey $ipnsKey = null)
    {
        $this->id = $id;
        $this->ipnsKey = $ipnsKey; 
    }

    public static function fromProto(\Bloock\PublishResponse $res): PublishResponse
    {
        return new PublishResponse($res->getId(), IpnsKey::fromProto($res->getIpnsKey()));
    }
}

File: languages/python/bloock/entity/availability/publish_response.py
from __future__ import annotations

import bloock._bridge.proto.bloock_availability_pb2 as proto
from bloock.entity.availability.ipns_key import IpnsKey


class PublishResponse:
    """
    Represents an object with a the publish response attributes.
    """
    id = None
    ipns_key = None

    def __init__(self, id: str, ipns_key: IpnsKey = None) -> None:
        """
        Creates an IpnsKey instance with a key uuid identifier.
        :type id: object
        :type ipns_key: object
        :rtype: object
        """
        self.id = id
        if isinstance(ipns_key, IpnsKey):
            self.ipns_key = ipns_key

    @staticmethod
    def from_proto(res: proto.PublishResponse) -> PublishResponse:
        return PublishResponse(res.id, IpnsKey.from_proto(res.ipns_key))